0_RootFS/gcc_sources.jl

```diff
@@ -159,8 +170,14 @@ function gcc_sources(gcc_version::VersionNumber, compiler_target::Platform; kwar
         binutils_sources = [
             GitSource("https://github.com/tpoechtrager/apple-libtapi.git",
                       "a66284251b46d591ee4a0cb4cf561b92a0c138d8"),
+            # GitSource("https://github.com/tpoechtrager/cctools-port.git",
+            #           "634a084377ee2e2932c66459b0396edf76da2e9f"),
+            # # 1300.6.5; API: 1.6.0
+            # GitSource("https://github.com/tpoechtrager/apple-libtapi.git",
+            #           "aa37c11ad1a817248c9d1578ac99e133875b4eb5"),
+            # cctools-877.8-ld64-253.9-1
             GitSource("https://github.com/tpoechtrager/cctools-port.git",
-                      "634a084377ee2e2932c66459b0396edf76da2e9f"),
+                      "2f128d5d369b688f7a3e872254c2079b87733c0a"),
```

Comment:
Reminder to self: don't unconditionally change the version here, we need to keep old build work as they are, this needs to be guarded by GCC version.

Reply:
Now I'm extremely confused: https://github.com/tpoechtrager/cctools-port/commit/634a084377ee2e2932c66459b0396edf76da2e9f was from 2020, https://github.com/tpoechtrager/cctools-port/commit/2f128d5d369b688f7a3e872254c2079b87733c0a is from 2016, why going back?

Reply:
This is also the cause of https://github.com/JuliaPackaging/Yggdrasil/pull/10132#discussion_r1901931624 and https://github.com/JuliaPackaging/Yggdrasil/pull/10132#discussion_r1901917587: newer versions of cctools-port don't have a problem with `sys/cdefs.h`.  If it wasn't clear, they don't do releases anymore, they have branches.  I'm trying to find a version, actually ***newer*** than https://github.com/tpoechtrager/cctools-port/commit/634a084377ee2e2932c66459b0396edf76da2e9f, which builds without too much hassle.

Reply:
I was able to compile https://github.com/tpoechtrager/cctools-port/tree/1010.6-ld64-951.9 successfully, but I'm considering updating the RootFS to newer Alpine version first, because we need [`libdispatch`](https://pkgs.alpinelinux.org/package/v3.17/community/x86_64/libdispatch) and [`libdispatch-dev`](https://pkgs.alpinelinux.org/package/v3.17/community/x86_64/libdispatch-dev), which are only in Alpine v3.17+